src/tests/ftest/deployment/server_rank_failure.py

```diff
@@ -154,8 +171,38 @@ def verify_rank_failure(self, ior_namespace):
         time.sleep(5)
 
         # 3. While IOR is running, kill all daos_engine on a non-access-point node
-        engine_kill_host = self.hostlist_servers[1]
-        self.kill_engine(engine_kill_host=engine_kill_host)
+        ex_env = self.params.get("ex_env", "/run/*")
+        if ex_env == "ci":
+            # Original implementation. Need to be changed to kill one engine process from two
+            # different nodes in two different groups.
+            engine_kill_host = self.hostlist_servers[1]
+            self.kill_engine(engine_kill_host=engine_kill_host)
+        else:
+            # Workaround for Aurora. Stop up to two ranks in different groups using
+            # "dmg system stop"
+            ranks = self.server_managers[0].ranks
+            self.log.info("Ranks = %s", ranks)
+            # Create a list of list that contains the ranks in each group. e.g.,
+            # First element is a list of ranks for group 1 nodes, or the list of nodes in
+            # the first element of host_groups.
+            rank_groups = defaultdict(list)
+            for rank, host in ranks.items():
+                for group in range(8):
+                    if host in HOST_GROUPS[group]:
+                        rank_groups[group].append(rank)
```

Comment:
And then here we don't have to hardcode `8` or even need to check anything at all
```suggestion
                rank_groups[aurora_host_to_group(host)].append(rank)
```

Reply:
hmm, I'm not sure what you mean here. What is `aurora_host_to_group`? Is it already implemented?

Reply:
Looks like my previous comment didn't go through. Let me repost